ee/vellum_ee/workflows/display/nodes/vellum/base_node.py

```diff
@@ -26,6 +26,19 @@ def serialize(self, display_context: WorkflowDisplayContext, **kwargs: Any) -> J
         node = self._node
         node_id = self.node_id
 
+        attributes: JsonArray = []
+        for name in node.__dict__.keys():
+            if name not in BaseNode.__dict__:
```

Comment:
[nit] you should be able to do `for attribute in node:` and `attribute` will be a `NodeReference` (a descriptor that references an attribute)

Reply:
@dvargas92495 I did this but I had to add a cast because `attribute.instance` is not typed as a `BaseDescriptor`

Reply:
That's because it shouldn't be - it can be Any:
`foo = "string"` -> constant values
`foo = MyNode.Outputs.foo` -> node reference